tests/prompts_test.py

```diff
@@ -0,0 +1,362 @@
+QWEN_MODEL = CONFIG_DATA.get("QWEN_MODEL", "")
+QWEN_ACCURACY = CONFIG_DATA.get("QWEN_ACCURACY", 0.65)
+QWEN_PROMPT_TEMPLATE = CONFIG_DATA.get("QWEN_PROMPT_TEMPLATE", "")
+
+
+class PromptTests(unittest.TestCase):
+    @staticmethod
+    def read_testcases_from_csv(file_path: str) -> list:
+        """
+        Reads a CSV file and returns a list of tuples (index, label, specification).
+
+        Args:
+            file_path (str): Path to the CSV file with test cases.
+
+        Returns:
+            list: List of tuples (index, label, specification).
+        """
+        result = []
+
+        try:
+            with open(file_path, "r") as csvfile:
+                reader = csv.reader(csvfile)
+                for row in reader:
+                    index, label, specification = row
+                    result.append((index.strip(), label.strip(), specification.strip()))
+        except FileNotFoundError:
+            raise FileNotFoundError(f"File not found in path: {file_path}")
+        return result[1:]
+
+    @staticmethod
+    def save_response_to_csv(data: list[tuple], llm_name: str) -> None:
+        """
+        Writes a list of tuples to a CSV file.
+
+        Args:
+            data (list): List of tuples (
+                (
+                "Circle Number",
+                "Test Number",
+                "Expected Label",
+                "LLM Result",
+                "LLM Step by Step Reasoning",
+                "Prompt Template",
+                "INSTRUCTION",
+                "OPTIONS",
+                "LLM"
+                )
+            )
+            llm_name (str): name of LLM
+        """
+
+        now = datetime.now(pytz.timezone("Europe/Kiev")).strftime("%Y-%m-%d-%H-%M-%S")
+
+        with open(f"tests_results/{llm_name}-{now}.csv", "w", newline="") as csvfile:
+            writer = csv.writer(csvfile)
+            writer.writerow(
+                [
+                    "Circle Number",
+                    "Test Number",
+                    "Expected Label",
+                    "LLM Result",
+                    "LLM Step by Step Reasoning",
+                    "Prompt Template",
+                    "INSTRUCTION",
+                    "OPTIONS",
+                    "LLM",
+                ]
+            )
+            for row in data:
+                writer.writerow(row)
+
+    def setUp(self):
+        self.testcases = self.read_testcases_from_csv("tests/testcases.csv")
+
+    def test_gpt_prompt(self):
+        match_count = 0
+        failed_testcases = ""
+        result_testing_tuples = []
+        client = OpenAI(
+            base_url=OPENROUTER_BASE_URL,
+            api_key=OPENROUTER_API_KEY,
+        )
+        for index_of_circle in range(
+            RETRY_COUNT
+        ):  # Retry tests for correct evaluation precision
+            for index_of_testcase, expected_result, specification in self.testcases:
+                with self.subTest(i=index_of_testcase):  # subTest for every llm call
```

Comment:
it doesn't look like good approach to write separate test for each LLM. This should be configurable from config, so when we add or remove LLM from config, we shoouldn't need to rewrite the test.

Reply:
I rewrote tests without code duplication. I couldn't do it before because we had different call structures and instruction for different LLMs. But now we have one function for testing and "MODELS_CONFIG" list with configs that we want to test. We can configure tests for different llms by adding them to this list in code, by such way:
![image](https://github.com/user-attachments/assets/541faf8f-257d-488f-bdd1-318d98c5dd31)